apps/src/dance/Dance.js

```diff
@@ -103,6 +103,10 @@ Dance.prototype.injectStudioApp = function (studioApp) {
  * @param {!Dancelab} config.level
  */
 Dance.prototype.init = function (config) {
+  // Only send LEVEL_ACTIVITY event for students or non-authenticated users
+  if (getStore().getState().currentUser?.userType !== 'teacher') {
```

Comment:
There are a lot of the `getStore().getState().currentUser?.userType !== 'teacher'` repeated all over. I could abstract this into a function, but I'm not sure where to put it. Some kind of utils or shared package.